src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -2785,5 +2837,1166 @@ private void AssertValid()
                 Debug.Assert(_sign > int.MinValue);
             }
         }
+
+        //
+        // IAdditionOperators
+        //
+
+        /// <inheritdoc cref="IAdditionOperators{TSelf, TOther, TResult}.op_Addition(TSelf, TOther)" />
+        static BigInteger IAdditionOperators<BigInteger, BigInteger, BigInteger>.operator checked +(BigInteger left, BigInteger right) => left + right;
+
+        //
+        // IAdditiveIdentity
+        //
+
+        /// <inheritdoc cref="IAdditiveIdentity{TSelf, TResult}.AdditiveIdentity" />
+        static BigInteger IAdditiveIdentity<BigInteger, BigInteger>.AdditiveIdentity => Zero;
+
+        //
+        // IBinaryInteger
+        //
+
+        /// <inheritdoc cref="IBinaryInteger{TSelf}.DivRem(TSelf, TSelf)" />
+        public static (BigInteger Quotient, BigInteger Remainder) DivRem(BigInteger left, BigInteger right)
+        {
+            BigInteger quotient = DivRem(left, right, out BigInteger remainder);
+            return (quotient, remainder);
+        }
+
+        /// <inheritdoc cref="IBinaryInteger{TSelf}.LeadingZeroCount(TSelf)" />
+        public static BigInteger LeadingZeroCount(BigInteger value)
+        {
+            value.AssertValid();
+
+            if (value._bits is null)
+            {
+                return int.LeadingZeroCount(value._sign);
+            }
+
+            // When the value is positive, we just need to get the lzcnt of the most significant bits
+            // Otherwise, we're negative and the most significant bit is always set.
+
+            return (value._sign >= 0) ? uint.LeadingZeroCount(value._bits[^1]) : 0;
+        }
+
+        /// <inheritdoc cref="IBinaryInteger{TSelf}.PopCount(TSelf)" />
+        public static BigInteger PopCount(BigInteger value)
+        {
+            value.AssertValid();
+
+            if (value._bits is null)
+            {
+                return int.PopCount(value._sign);
+            }
+
+            ulong result = 0;
+
+            if (value._sign >= 0)
+            {
+                // When the value is positive, we simply need to do a popcount for all bits
+
+                for (int i = 0; i < value._bits.Length; i++)
+                {
+                    uint part = value._bits[i];
+                    result += uint.PopCount(part);
```

Comment:
Would it be worth reinterpreting _bits as ulongs and doing a PopCount on each of those (plus any additional uint that remains)?

Reply:
It would probably be better to use `nuint` (so 32-bit perf stays good), but there is a more general rewrite of `BigInteger` planned at some point and I think it would be better to do that then rather than complicate the current logic more.
* The general goal of the rewrite is to make `BigInteger` use `nuint[]` (rather than `uint[]`) and make it two's complement, rather than one's complement representation